### PR TITLE
audiounit: expect error when set device info.

### PR DESCRIPTION
**Pull request by achronop**
Return cubeb error status from `audiounit_set_device_info` since `audiounit_get_default_device_id` can fail.

Error found in the following crash id:
https://crash-stats.mozilla.com/report/index/920b342c-e152-471d-ae16-fcb800170919

### CODE REVIEW

File: src/cubeb_audiounit.cpp [hunk 1]
```diff
@@ -645,12 +650,14 @@ audiounit_reinit_stream(cubeb_stream * stm, device_flags_value flags)
      * default system device change. In both cases cubeb switch on the new default
      * device. This is considered the most expected behavior for the user. */
     if (flags & DEV_INPUT) {
-      audiounit_set_device_info(stm, 0, INPUT);
+      r = audiounit_set_device_info(stm, 0, INPUT);
+      assert(r == CUBEB_OK);
```
Comment (padenot) [inline src/cubeb_audiounit.cpp]: Is it ok to assert here? This can happen, right?

File: src/cubeb_audiounit.cpp [hunk 2]
```diff
@@ -645,12 +650,14 @@ audiounit_reinit_stream(cubeb_stream * stm, device_flags_value flags)
...
     }
     /* Always use the default output on reinit. This is not correct in every case
      * but it is sufficient for Firefox and prevent reinit from reporting failures.
      * It will change soon when reinit mechanism will be updated. */
-    audiounit_set_device_info(stm, 0, OUTPUT);
+    r = audiounit_set_device_info(stm, 0, OUTPUT);
+    assert(r == CUBEB_OK);
```
Comment (padenot) [inline src/cubeb_audiounit.cpp]: Same.

File: src/cubeb_audiounit.cpp (hunk 1 above)
Comment (achronop) [reply src/cubeb_audiounit.cpp]: I see the failure reported on init only. I believe it has something to do with the initial configuration. Probably, if the init works it does not fail on reinit. I left it like that in order to confirm (or not) my theory. I can change it if you insist.

Comment (padenot) [reply src/cubeb_audiounit.cpp]: Ok. We have been annoying people recently with assert crashes in Nightly. Hopefully this will be useful.

**Pull request merged**